presto-main/src/main/java/com/facebook/presto/sql/gen/InputReferenceCompiler.java

```diff
@@ -58,8 +58,13 @@ public BytecodeNode visitInputReference(InputReferenceExpression node, Scope sco
 
         BytecodeExpression block = blockResolver.apply(scope, field);
         BytecodeExpression position = positionResolver.apply(scope, field);
-        Variable wasNullVariable = scope.getVariable("wasNull");
 
+        return generateParameter(callSiteBinder, scope, type, block, position);
+    }
+
+    public static BytecodeNode generateParameter(CallSiteBinder callSiteBinder, Scope scope, Type type, BytecodeExpression block, BytecodeExpression position)
```

Comment:
Move this utility method into `BytecodeUtils`.  Make this refactor a separate commit.

The name `generateParameter` seems too broad. But I don't have a good suggestion here. I am thinking somthing like `loadValueFromBlockAndPosition` but seems too long?  @haozhun